nix/download.tt

```diff
@@ -4,23 +4,30 @@
 
 <p>The latest and recommended version of Nix is
 <strong>[%latestNixVersion%]</strong>. The quickest way to install it
-on Linux and Mac OS X is to run the following in a shell (as a user
+on Linux and macOS is to run the following in a shell (as a user
 other than <tt>root</tt>):
 
-<pre><span class="muted">$</span> curl <a href="[%root%]nix/install">https://nixos.org/nix/install</a> | sh</pre>
+<pre>
+<span class="command">curl -o install-nix-[%latestNixVersion%] https://nixos.org/nix/install</span>
+<span class="command">echo "[%nixInstallHashsha256%]  install-nix-[%latestNixVersion%]" | (shasum -p 256 || sha256sum -c) 2> /dev/null</span>
+<span class="command">sh install-nix-[%latestNixVersion%]</span>
+</pre>
 
-Make sure to follow the instructions output by the script.
+<p>Make sure to follow the instructions output by the script.</p>
 
-You may want to verify the integrity of the installation script using GPG:
+<p>You may want to verify the integrity of the installation script
+using GPG:</p>
 
 <pre>
-<span class="muted">$</span> curl -O <a href="[%root%]nix/install">https://nixos.org/nix/install</a>
-<span class="muted">$</span> curl -O <a href="[%root%]nix/install.sig">https://nixos.org/nix/install.sig</a>
-<span class="muted">$</span> gpg2 --verify ./install.sig
-<span class="muted">$</span> sh ./install
+<span class="command">curl -o install-nix-[%latestNixVersion%] https://nixos.org/nix/install</span>
+<span class="command">curl -o install-nix-[%latestNixVersion%].sig https://nixos.org/nix/install.sig</span>
+<span class="command">gpg2 --recv-keys B541D55301270E0BCF15CA5D8170B4726D7198DE</span>
+<span class="command">gpg2 --verify ./install-nix-[%latestNixVersion%].sig</span>
```

Comment:
Not to expand the install steps much, but with this line it's important that the two curl commands above are executed as-is (specifically, the two output files are named as they're indicated.

`gpg2 --verify`, if only given the `.sig` file, looks for a file with the _same name_, less the `.sig` to be the "signed file".

Though `man gpg2` will tell anyone that, it may be an unexpected error and "fun" install process if the end-user is unaware and decides to curl into a different filename for either step...